rootio/streamers.go

```diff
@@ -72,10 +72,8 @@ func (tsi *tstreamerInfo) MarshalROOT(w *WBuffer) (int, error) {
 	w.WriteU32(tsi.chksum)
 	w.WriteI32(tsi.clsver)
 
-	tsi.objarr.arr = make([]Object, tsi.objarr.Len())
-	for _, obj := range tsi.objarr.arr {
-		w.WriteObjectAny(obj)
-	}
+	tsi.objarr.arr = make([]Object, len(tsi.elems))
```

Comment:
yes. but then `tsi.objarr.arr` is just a slice with `n` elements of type `rootio.Object` and value "the zero value of `rootio.Object`" (which is `nil`.)

here:
```go
func (tsi *tstreamerInfo) MarshalROOT(w *WBuffer) (int, error) {
        if w.err != nil {
                return 0, w.err
        }

        pos := w.Pos()
        w.WriteVersion(tsi.rvers)
        tsi.named.MarshalROOT(w)
        w.WriteU32(tsi.chksum)
        w.WriteI32(tsi.clsver)

        if len(tsi.elems) > 0 {
                tsi.objarr.arr = make([]Object, len(tsi.elems))
                for i, v := range tsi.elems {
                        tsi.objarr.arr[i] = v
                }
        }
        w.WriteObjectAny(tsi.objarr)
        tsi.objarr.arr = nil

        return w.SetByteCount(pos, "TStreamerInfo")
}
```

Reply:
I don't understand why is tsi.objarr.arr nil-ed again at the end of MarshalROOT ?

Reply:
this is to prevent a possible hysteresis effect, (marginally) reduce memory usage and make sure we always have a single point of truth (SPOT) for the streamer elements informations.